src/main/scala/com/datastax/gatling/plugin/request/GraphRequestAction.scala

```diff
@@ -69,7 +71,18 @@ class GraphRequestAction(val name: String,
       ThroughputVerifier.checkForGatlingOverloading(session, gatlingTimingSource)
       GatlingResponseTime.startedByGatling(session, gatlingTimingSource)
     }
-    val stmt = dseAttributes.statement.buildFromSession(session)
+
+    // Attempt to generate a graph statement from our parameters, propagating any uncaught exceptions after
+    // continuing the chain with a failed session
+    val stmt: Validation[GraphStatement] = try {
+      dseAttributes.statement.buildFromSession(session)
+    } catch {
+      case e: Throwable => {
+        logger.error("Failed to generate GraphStatement", e)
+        next ! session.markAsFailed
+        throw e
+      }
+    }
```

Comment:
You wrapped the call `dseAttributes.statement.buildFromSession(session)` with a try catch.  But according to the type system, this call should return a `Validation[GraphStatement]`.  A `Validation` is a Gatling structure that represents either a `Success` or `Failure` (http://gatling.io/docs/current/session/validation/).  So really, this case of an uncaught exception should have been handled before we return to GraphRequestAction.

I think that the fix should be located in [`DseGraphStatements.scala:54`](https://github.com/datastax/gatling-dse-plugin/blob/98443acf94189e4121c7391d925e2cf0239fe181/src/main/scala/com/datastax/gatling/plugin/model/DseGraphStatements.scala#L54) instead of here.  Proposition: replace L54 in `DseGraphStatements` by `safely()(lambda(gatlingSession).success)`.  This will use a built-in Gatling function that does exactly what you want, while also satisfying the type system.

Reply:
Thanks for the insight.

Gatling's `safely` is a lot more concise and idiomatic than `try-catch`. There's one thing that makes me uneasy about it though -- I don't see a way to log the stacktrace of the exception caught by `safely` before it goes out of scope.  For a user-provided lambda, there could be cases where the exception stacktrace is useful, potentially significantly more useful than the exception type/message (e.g. NPE).

I just pushed a new commit that moves exception handling down to DseGraphStatements.scala:54 as directed, but uses `Try { ... } match { ... }` instead of `safely`.  It's now similar to some of its peer implementations inside `DseGraphStatements.scala` that also already used similar `Try { ... } match { ... }` constructs, and I can still see lambda traces in the logs.

What do you think?

Theoretically, pushing down exception handling like this could allow some other implementation of `buildFromSession` to induce a hang, if that implementation allows uncaught exceptions to propagate out to the caller; but it looks like some of the other implementations in `DseGraphStatements` already go out of their way to prevent that, and I'm assuming your comment means that's the general rule.

Reply:
>  I don't see a way to log the stacktrace of the exception caught by safely before it goes out of scope. 

That's a good reason to use a try/catch block, indeed.  +1 for that approach

> Theoretically, pushing down exception handling like this could allow some other implementation of `buildFromSession` to induce a hang

I agree with you.  I also wondered whether to keep the extra try/catch in GraphRequestAction, just to be 100% future proof.  But I think the way to go is to honor the type system and make it clear that errors are prevented.

Reply:
> I agree with you. I also wondered whether to keep the extra try/catch in GraphRequestAction, just to be 100% future proof. But I think the way to go is to honor the type system and make it clear that errors are prevented.

Sounds good, I'll stick to the pushdown approach and update the tests to match.  Thank you for addressing this.